### PR TITLE
Add confluent-cli cask v2.3.1

### PR DESCRIPTION
**Pull request by florianmutter**
After making all changes to a cask, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [x] `brew audit --new-cask <cask>` worked successfully.
- [x] `brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

File: Casks/confluent-cli.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  sha256 "7fe81e0d007df112a6679178ac302a7c414b3f1417d69046b2e773dd7d669691"
+
+  url "https://s3-us-west-2.amazonaws.com/confluent.cloud/confluent-cli/archives/#{version}/confluent_v#{version}_darwin_amd64.tar.gz",
+      verified: "https://s3-us-west-2.amazonaws.com/confluent.cloud/confluent-cli/"
+  name "confluent-cli"
+  desc "Enables developers to manage Confluent Cloud or Confluent Platform"
+  homepage "https://docs.confluent.io/confluent-cli/current/overview.html"
+
+  livecheck do
+    skip "No version information available"
```
Comment (SMillerDev) [inline Casks/confluent-cli.rb:12]: https://docs.confluent.io/confluent-cli/current/release-notes.html has version information

Comment (SMillerDev) [reply Casks/confluent-cli.rb:12]: Alternatively https://s3-us-west-2.amazonaws.com/confluent.cloud/confluent-cli/archives/latest/confluent_latest_darwin_amd64.tar.gz also reveals the version in the headers

**Review by SMillerDev — COMMENTED** [review]
Please add a [`zap` stanza](https://docs.brew.sh/Cask-Cookbook#stanza-zap) here. The [CreateZap](https://github.com/nrlquaker/homebrew-createzap) tool can help you with this.

**Comment by florianmutter** [comment]
Thanks @miccal for adding the livecheck. I added the zap stanza. @SMillerDev please have a look again

**Comment by miccal** [comment]
Thank you @florianmutter.

**Pull request merged**